ci-operator/step-registry/openshift/hac-dev/e2e/openshift-hac-dev-e2e-commands.sh

```diff
@@ -0,0 +1,126 @@
+CYPRESS_RP_HAC=$(cat /usr/local/ci-secrets/github/report-portal-token-hac)
+CYPRESS_RP_HAC=$(cat /usr/local/ci-secrets/github/report-portal-token-hac)
+
+#QONTRACT
+export QONTRACT_PASSWORD QONTRACT_USERNAME QONTRACT_BASE_URL
+QONTRACT_PASSWORD=$(cat /usr/local/ci-secrets/github/QONTRACT_PASSWORD)
+QONTRACT_USERNAME=$(cat /usr/local/ci-secrets/github/QONTRACT_USERNAME)
+QONTRACT_BASE_URL="https://app-interface.devshift.net/graphql"
+
+#ephemeral-bot
+export OC_LOGIN_TOKEN OC_LOGIN_SERVER
+OC_LOGIN_TOKEN=$(cat /usr/local/ci-secrets/ephemeralbot/oc-login-token)
+OC_LOGIN_SERVER=$(cat /usr/local/ci-secrets/ephemeralbot/oc-login-server)
+
+echo "Preparing bonfire config"
+CONFIG_DIR=$(mktemp -d)
+cat > "$CONFIG_DIR/config.yaml" << EOF
+# Bonfire deployment configuration
+# Defines where to fetch the file that defines application configs
+appsFile:
+  host: gitlab
+  repo: insights-platform/cicd-common
+  path: bonfire_configs/ephemeral_apps.yaml
+# (optional) define any apps locally. An app defined here with <name> will override config for app
+# <name> in above fetched config.
+apps:
+- name: insights-ephemeral
+  components:
+    - name: frontend-configs
+      host: github
+      repo: redhat-hac-qe/frontend-configs
+      path: deploy/deploy.yaml
+EOF
+
+echo "HAC_KC_SSO_URL & OC_LOGIN_SERVER"
+echo "$HAC_KC_SSO_URL"
+echo "$OC_LOGIN_SERVER"
+
+# Get bonfire helper scripts and python venv. Set GIT_COMMIT and IMAGE_TAG
+CICD_URL=https://raw.githubusercontent.com/RedHatInsights/bonfire/master/cicd
+curl -s $CICD_URL/bootstrap.sh > .cicd_bootstrap.sh && source .cicd_bootstrap.sh
+
+# Note: PoC will be cleaned up with Bonfire changes
+# Get a namespace in the eph cluster and set vars accordingly
+NAMESPACE=$(bonfire namespace reserve)
+ENV_NAME=env-${NAMESPACE}
+oc project ${NAMESPACE}
+HOSTNAME=$(oc get feenv ${ENV_NAME} -o json | jq ".spec.hostname" | tr -d '"')
+
+# Temp: setup proxy and patch SSO for devsandbox
+oc patch feenv ${ENV_NAME} --type merge  -p '{"spec":{"sso": "'$HAC_KC_SSO_URL'" }}'
+oc process -f tmp/hac-proxy.yaml -n ${NAMESPACE} -p NAMESPACE=${NAMESPACE} -p ENV_NAME=${ENV_NAME} -p HOSTNAME=${HOSTNAME} | oc create -f -
+
+# Deploy hac-dev with PR git ref and mainline hac-core ref
+bonfire deploy -c "$CONFIG_DIR/config.yaml" \
+        hac \
+        --frontends true \
+        --source=appsre \
+        --clowd-env ${ENV_NAME} \
+        --set-image-tag ${CHROME_IMAGE}=${CHROME_TAG} \
+        --namespace ${NAMESPACE}
+
+# Call the keycloak API and add a user
+B64_USER=$(oc get secret ${ENV_NAME}-keycloak -o json | jq '.data.username'| tr -d '"')
+B64_PASS=$(oc get secret ${ENV_NAME}-keycloak -o json | jq '.data.password' | tr -d '"')
+# These ENVs are populated in the Jenkins job by Vault secrets
+python tmp/keycloak.py $HAC_KC_SSO_URL $HAC_KC_USERNAME $HAC_KC_PASSWORD $B64_USER $B64_PASS $HAC_KC_REGISTRATION
+
+cd integration-tests || exit 1
+npm install
+npm install typescript
```

Comment:
I would rather add that to the dependencies directly.

Reply:
Yes It can be done, but for that we have to raise a separate PR in HAC-dev to have typescript in it and later it can be removed 🤔